tls/s2n_server_new_session_ticket.c

```diff
@@ -74,3 +75,112 @@ int s2n_server_nst_send(struct s2n_connection *conn)
 
     return 0;
 }
+
+static S2N_RESULT s2n_generate_ticket_lifetime(struct s2n_connection *conn, uint32_t *ticket_lifetime) 
+{
+    ENSURE_REF(conn);
+    ENSURE_MUT(ticket_lifetime);
+
+    uint32_t key_lifetime_in_secs = conn->config->decrypt_key_lifetime_in_nanos / ONE_SEC_IN_NANOS;
+    uint32_t session_lifetime_in_secs = conn->config->session_state_lifetime_in_nanos / ONE_SEC_IN_NANOS;
+    uint32_t key_and_session_min_lifetime = MIN(key_lifetime_in_secs, session_lifetime_in_secs);
+    /** 
+     *= https://tools.ietf.org/rfc/rfc8446#section-4.6.1
+     *# Servers MUST NOT use any value greater than
+     *# 604800 seconds (7 days).
+     **/
```

Comment:
Not a blocker, but it might be nice to annotate each of these "generate" methods with the description of that field, even though they don't contain MUSTs. So for ticket lifetime, you'd do:
```
/** 
 *= https://tools.ietf.org/rfc/rfc8446#section-4.6.1
 *# Indicates the lifetime in seconds as a 32-bit
 *# unsigned integer in network byte order from the time of ticket
 *# issuance. 
 **/
 static S2N_RESULT s2n_generate_ticket_lifetime(struct s2n_connection *conn, uint32_t *ticket_lifetime) 
```